src/main/java/mServer/crawler/sender/MediathekArd.java

```diff
@@ -28,488 +29,563 @@
+				s = s.replace("\"", "");
+				if (s.contains(",")) {
+					String[] ar = s.split(",");
+					for (String ss : ar) {
+						if (ss.startsWith("http")) {
+							ret = ss;
+						}
+					}
+				}
+				if (!ret.isEmpty()) {
+					break;
+				}
+			}
+			// if (!ret.isEmpty()) {
+			// System.out.println("gefunden!!");
+			// }
+			liste.clear();
+			return ret;
+		}
+
+		private void filmSuchen_old(String urlSendung, String thema, String titel, long dauer, String datum,
+				String zeit) {
+			// für ganz alte Sachen:
+			/*
+			 * </li><li
+			 * data-ctrl-21282662-28534346-trigger="{&#039;xt_obj&#039;:{&#039;
+			 * href&#039;
+			 * :&#039;http://cdn-vod-ios.br.de/i/mir-live/bw1XsLzS/bLQH/
+			 * bLOliLioMXZhiKT1/uLoXb69zbX06/MUJIuUOVBwQIb71S/
+			 * bLWCMUJIuUOVBwQIb71S/_2rp9U1S/_-J
+			 * S/_-4y5H1S/bc3757a9-1cfd-4aaa-a963-51e9fd0095f2_,0,A,B,E,C,.mp4.
+			 * csmil/master.m3u8?__b__=200&#039;,&#039;target&#039;:&#039;_self&
+			 * #039;},&#039;redirect&#0 39;:{&#039;name&#039;:&#039;br&#039;}}">
+			 */
+			try {
+				meldung(urlSendung);
+				seite2 = getUrl.getUri_Utf(SENDERNAME, urlSendung, seite2, "");
+				if (seite2.length() == 0) {
+					Log.errorLog(612031478, "Leere Seite: " + urlSendung);
+					return;
+				}
+				String url = seite2.extract("</li><li data-ctrl-", "http://", ".m3u8");
+				if (!url.isEmpty()) {
+					url = "http://" + url + ".m3u8";
+					// System.out.println(url);
+				}
+				if (!url.isEmpty()) {
+					String beschreibung = beschreibung(urlSendung);
+					DatenFilm f = new DatenFilm(SENDERNAME, thema, urlSendung, titel, url, ""/* urlRtmp */, datum, zeit,
+							dauer, beschreibung);
+					addFilm(f);
+				} else {
+					Log.errorLog(974125698, "keine URL: " + urlSendung);
+				}
+			} catch (Exception ex) {
+				Log.errorLog(102054784, ex);
+			}
+		}
+
+		private String beschreibung(String strUrlFeed) {
+			seite3 = getUrl.getUri_Utf(SENDERNAME, strUrlFeed, seite3, "");
+			if (seite3.length() == 0) {
+				Log.errorLog(784512036, "Leere Seite: " + strUrlFeed);
+				return "";
+			}
+			return seite3.extract("<p class=\"subtitle\">", "<p class=\"teasertext\" itemprop=\"description\">", "<");
+		}
+
+	}
+
+	private boolean isAllTextsEmpty(String... aTexts) {
+		Boolean isAllTextsEmpty = null;
+		for (String text : aTexts) {
+			isAllTextsEmpty = (isAllTextsEmpty == null || isAllTextsEmpty) && StringUtils.isEmpty(text);
+		}
+		return isAllTextsEmpty == null || isAllTextsEmpty;
+	}
+
+	/**
+	 * Searches the Seite for a quality auto to get a M3U8 URL. If the URL is
+	 * from WRD it searches for the URLs of the MP4 files.
+	 *
+	 * @param aSeiteStringExtractor
+	 *            The Seite.
+	 * @return A Map containing the URLs and Qualities which was found. An empty
+	 *         Map if nothing was found.
+	 */
+	private Map<Qualities, String> searchForUrlsWithM3U8(final MSStringBuilder aSeiteStringExtractor) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move this method into "ThemaLaden". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS3398)